pkg/plugins/golang/v3/scaffolds/internal/templates/dockerfile.go

```diff
@@ -41,6 +41,8 @@ func (f *Dockerfile) SetTemplateDefaults() error {
 const dockerfileTemplate = `# Build the manager binary
 FROM golang:1.17 as builder
 
+ARG GOARCH=amd64
```

Comment:
If we change the value informed by the ` --build-arg GOARCH=$(shell go env GOARCH) ` will it work or we will build to amd64 always? 
Should we not create a parameter to receive the OS as well?